components/DatePicker.tsx

```diff
@@ -0,0 +1,198 @@
+  const [isAndroidPickerVisible, setIsAndroidPickerVisible] = useState(false)
+
+  useEffect(() => {
+    setFromTime(initialFromTime)
+    setToTime(initialToTime)
+  }, [initialFromTime, initialToTime])
+
+  const formatTime = useCallback((date: Date) => {
+    return dayjs(date).format("h:mm")
+  }, [])
+
+  const openTimePicker = (type: "FROM" | "TO") => {
+    setActiveTimeType(type)
+    setTempTime(type === "FROM" ? fromTime : toTime)
+
+    if (Platform.OS === "android") {
+      setIsAndroidPickerVisible(true)
+    } else {
+      setIsBottomDrawerVisible(true)
+    }
+  }
+
+  const closeDrawer = useCallback(() => setIsBottomDrawerVisible(false), [])
+
+  const handleTimeChange = (
+    event: DateTimePickerEvent,
+    selectedDate?: Date,
+  ) => {
+    if (event.type === "set" && selectedDate) {
+      setTempTime(selectedDate)
+      if (Platform.OS === "android") {
+        saveTime(selectedDate)
+      }
+    }
+    setIsAndroidPickerVisible(false)
+  }
+
+  const saveTime = useCallback(
+    (newTime?: Date) => {
+      if (!newTime) return
+
+      if (activeTimeType === "FROM") {
+        setFromTime(newTime)
+        if (dayjs(newTime).isAfter(dayjs(toTime))) {
+          setToTime(dayjs(newTime).add(1, "hour").toDate())
+        }
+      } else {
+        setToTime(newTime)
+      }
+
+      onSave(
+        activeTimeType === "FROM" ? newTime : fromTime,
+        activeTimeType === "TO" ? newTime : toTime,
+      )
+      closeDrawer()
+    },
+    [activeTimeType, fromTime, toTime, onSave, closeDrawer],
+  )
+
+  const getMinimumDate = useCallback(() => {
+    if (activeTimeType === "FROM") {
+      return new Date()
+    } else {
+      return fromTime
+    }
+  }, [activeTimeType, fromTime])
+
+  const getMaximumDate = useCallback(() => {
+    return dayjs().add(24, "hour").toDate()
+  }, [])
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.contentContainer}>
+        <View style={styles.column}>
+          <Text style={styles.label}>FROM</Text>
+          <TouchableOpacity onPress={() => openTimePicker("FROM")}>
+            <Text style={styles.timeText}>{formatTime(fromTime)}</Text>
+          </TouchableOpacity>
+        </View>
+        <View style={styles.column}>
+          <Text style={styles.label}>TO</Text>
+          <TouchableOpacity onPress={() => openTimePicker("TO")}>
+            <Text style={styles.timeText}>{formatTime(toTime)}</Text>
+          </TouchableOpacity>
+        </View>
+        <TouchableOpacity style={styles.closeButton} onPress={onClose}>
+          <CloseIcon color={theme.colors.black} />
+        </TouchableOpacity>
+      </View>
+      {Platform.OS === "android" && isAndroidPickerVisible && (
+        <DateTimePicker
+          value={tempTime}
+          mode="time"
+          is24Hour={false}
+          display="clock"
+          onChange={handleTimeChange}
+          minimumDate={getMinimumDate()}
+          maximumDate={getMaximumDate()}
+        />
```

Comment:
You can Isolate this component with all the Date logic and just use for as standalone for android and in the Drawer for IOS